projects/gnu.org/gcc/package.yml

```diff
@@ -0,0 +1,123 @@
+    patch -p1 <props/aarch64.diff
+
+    # skip multilib prefixes for best interop
+    sed -i.bak -e 's_m64=../lib64_m64=_' gcc/config/i386/t-linux64
+    rm gcc/config/i386/t-linux64.bak
+
+
+
+
+    ### /*** BOOTSTRAP BLOCK - REMOVE AFTER BOOTSTRAP ***/
+    if test "{{hw.platform}}" = "linux"; then
+      if command -v sudo; then
+        sudo apt-get --yes install gcc g++
+      else # assume no sudo = we can just run apt-get
+        apt-get --yes install gcc g++
+      fi
+    fi
+    ### /*** END BOOTSTRAP BLOCK ***/
+
+
+
+
+
+    mkdir build
+    cd build
+
+    # Note that env vars with spaces don't behave well, due to shells and life
+    # in general. So we handle those in script. Unless and until we decide to
+    # change to bash arrays, as in https://github.com/teaxyz/cli/pull/283,
+    # but that will have untold and unforseen consequences.
+
+    # convert ARGS to an array since we need space-including values
+    export ARGS=($ARGS --with-pkgversion="tea GCC {{version}}")
+    if test {{hw.platform}} = "linux"; then
+      export ARGS=("${ARGS[@]}" --with-boot-ldflags="-static-libstdc++ -static-libgcc $LDFLAGS")
+    fi
+
+    ../configure "${ARGS[@]}"
+    make --jobs {{ hw.concurrency }}
+    make install
+
+  env:
+    ARGS:
+      - --prefix={{ prefix }}
+      - --libdir={{ prefix }}/lib
+      - --enable-languages=c,c++,objc,obj-c++
+      - --with-bugurl="https://github.com/teaxyz/pantry.extra/issues"
+      - --disable-bootstrap
+      - --disable-nls
+    linux:
+      ARGS:
+        - --disable-multilib
+        - --enable-default-pie
+    darwin:
+      ARGS:
+        # Reliance on CLT hard path is yuck.
+        - --with-sysroot=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
+    darwin/aarch64:
+      ARGS:
+        # https://en.wikipedia.org/wiki/Darwin_(operating_system)#Darwin_16%E2%80%9319;_OS_X_rebranded_into_macOS
+        - --build=aarch64-apple-darwin20.0.0
+    darwin/x86-64:
+      ARGS:
+        - --build=x86_64-apple-darwin20.0.0
+
+test:
+  # FIXME: remove after bootstrap of gcc/binutils
+  dependencies:
+    gnu.org/binutils: '*'  # for `as`
+  script: |
+    gcc --version | grep -q "tea GCC {{version}}"
+    gcc -o test1 test.c -lgmp
+    ./test1
+    g++ -o test2 test.cc
+    test "$(./test2)" = "Hello, world!"
+
+provides:
+  # shouldn't need to support these.
+  # - bin/{{hw.arch}}-{{##MANUFACTURER}}-{{hw.platform}}{{##KERNEL-VERSION}}-c++
+  # - bin/{{hw.arch}}-{{##MANUFACTURER}}-{{hw.platform}}{{##KERNEL-VERSION}}-g++
+  # - bin/{{hw.arch}}-{{##MANUFACTURER}}-{{hw.platform}}{{##KERNEL-VERSION}}-gcc
+  # - bin/{{hw.arch}}-{{##MANUFACTURER}}-{{hw.platform}}{{##KERNEL-VERSION}}-gcc-{{version}}
+  # - bin/{{hw.arch}}-{{##MANUFACTURER}}-{{hw.platform}}{{##KERNEL-VERSION}}-gcc-ar
+  # - bin/{{hw.arch}}-{{##MANUFACTURER}}-{{hw.platform}}{{##KERNEL-VERSION}}-gcc-nm
+  # - bin/{{hw.arch}}-{{##MANUFACTURER}}-{{hw.platform}}{{##KERNEL-VERSION}}-gcc-ranlib
+  - bin/c++
```

Comment:
we shouldn't provide this. Instead (I think) `tea.xyz/gx/cc` does. This is how llvm does it too. Then we make the gx version able to use gcc or llvm based on which is in `PATH`.

tea provides `TEA_PKGS` now to help with this.